docs/specification/draft/basic/authentication.md

```diff
@@ -0,0 +1,385 @@
+Server Metadata protocol defined in
+[RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
+
+The discovery flow is illustrated below:
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant S as Server
+
+    C->>S: GET /.well-known/oauth-authorization-server
+    alt Discovery Success
+        S->>C: 200 OK + Metadata Document
+        Note over C: Use endpoints from metadata
+    else Discovery Failed
+        S->>C: 404 Not Found
+        Note over C: Fall back to default endpoints
+    end
+    Note over C: Continue with authentication flow
+```
+
+#### 2.3.1 Fallbacks for Servers without Metadata Discovery
+
+For servers that do not implement OAuth 2.0 Authentication Server Metadata, clients
+**MUST** use the following default endpoint paths relative to the server's base URL:
+
+| Endpoint               | Default Path | Description                          |
+| ---------------------- | ------------ | ------------------------------------ |
+| Authorization Endpoint | /authorize   | Used for authorization requests      |
+| Token Endpoint         | /token       | Used for token exchange & refresh    |
+| Registration Endpoint  | /register    | Used for dynamic client registration |
+
+Clients **MUST** first attempt to discover endpoints via the metadata document before
+falling back to default paths. When using default paths, all other protocol requirements
+remain unchanged.
+
+### 2.3 Client Registration Requirements
+
+MCP defines two distinct registration scenarios based on the redirect URI:
+
+#### 2.3.1 Localhost Redirect URIs
+
+When using localhost redirect URIs (http://localhost:{port} or http://127.0.0.1:{port}),
+clients:
+
+- Do **NOT** require dynamic registration
+- **MUST** use PKCE
+- **MAY** proceed directly to authorization
+- **MUST NOT** require client secrets
+
+This exception for localhost is explicitly supported by OAuth 2.1 for public clients and
+provides a secure flow through the combination of PKCE and localhost-only redirects.
+
+#### 2.3.2 Non-Localhost Redirect URIs
+
+For all other redirect URIs, dynamic client registration is **REQUIRED**. This provides a
+standardized way for clients to automatically register with new servers, which is crucial
+for MCP because:
+
+- Clients cannot know all possible servers in advance
+- Manual registration would create friction for users
+- It enables seamless connection to new servers
+- Servers can implement their own registration policies
+
+#### 2.3.3 Registration Process
+
+For non-localhost scenarios, clients **SHOULD** implement the OAuth 2.0 Dynamic Client
+Registration Protocol as defined in
+[RFC7591](https://datatracker.ietf.org/doc/html/rfc7591). This enables automatic
+registration with the server's registration endpoint to obtain the necessary client
+credentials.
+
+The registration request and response formats, along with all requirements and error
+handling, are defined in RFC7591.
```

Comment:
This section feels redundant (and with the "SHOULD," contradictory). If this is just meant to point to the RFC, can we link it in the above sections?